### PR TITLE
[FIX] account: Use absolute values when comparing negative amounts

### PR DESCRIPTION
**Pull request by grindtildeath**
match_total_amount_param on account.reconciliation.model is not
respected when paying a vendor bill through bank statement because
the amounts from both the statement line and the move line will
be negative.

Description of the issue/feature this PR addresses:

Auto-reconciliation between bank statement and
vendor bills is done without ref matching or amount
matching

Current behavior before PR:

1. On runbot, mark ‘Show full accounting features’ for you user.
2. Create a new Vendor bill using a new partner with an invoice line:
3. Validate the vendor bill, such an account move will be created:
Note that we don’t set any ref nor a label on the payable move line.
4. Check the reconciliation models, there must be only one looking as follows:Note that amount matching is marked with 100% and we use auto-validate.
5. Create a new bank statement with a to pay the vendor bill:
Note that the amount is LOWER than the vendor bill total and that the label (Dummy Label) does
NOT match the reference of the vendor bill nor the name on the move line.
6. Click on Reconcile and you will see the rainbowman....So reconciliation is done although the amount DOES NOT MATCH and the ref DOES NOT
MATCH.
Therefore, vendor bill is set as paid although it has NOT been paid completely.

Desired behavior after PR is merged:

Click on Reconcile and no reconciliation is done since the amount DOES NOT MATCH and the ref DOES NOT MATCH.
The vendor bill has to stay open as it's not completely paid.    

The problem lies on account.reconcile.model in _check_rule_propositions function:
https://github.com/odoo/odoo/blob/12.0/addons/account/models/
account_reconcile_model.py#L572-L578
As we’re paying a vendor bill (and not a customer invoice), both line_residual and total_residual are
negative, so the check must use absolute values in order to respect the Amount Matching %
parameter from the reconciliation model.

OPW-2179739
--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account/models/account_reconcile_model.py [hunk 1]
```diff
@@ -569,10 +569,10 @@ def _check_rule_propositions(self, statement_line, candidates):
         if float_is_zero(total_residual - line_residual, precision_rounding=line_currency.rounding):
             return True
 
-        if line_residual > total_residual:
-            amount_percentage = (total_residual / line_residual) * 100
+        if abs(line_residual) > abs(total_residual):
```
Comment (smetl) [inline addons/account/models/account_reconcile_model.py]: Most of the time, it's a very bad idea to use absolute values in accounting. In your case, you are adding a new bug when `line_residual` and `total_residual` are not sharing the same sign. I suggest something like:
```
line_residual_to_compare = line_residual if line_residual > 0.0 else -line_residual
total_residual_to_compare = total_residual if line_residual > 0.0 else -total_residual
...
```

Also, adapt your commit message.

Comment (smetl) [reply addons/account/models/account_reconcile_model.py]: Your commit should reflect what your are fixing, not the way you are doing it

Comment (grindtildeath) [reply addons/account/models/account_reconcile_model.py]: Hello @smetl 

I understand there could be an issue when `line_residual` and `total_residual` do not share the same sign, however, can you please explain how your proposal is different from using `abs` ?

Comment (smetl) [reply addons/account/models/account_reconcile_model.py]: @grindtildeath both are managing the sign regarding the statement line sign.

Example:
line_residual = -5
total_residual = 6

Using abs: 5 > 6 is False
Using my suggestion: 5 > -6 is True

Comment (grindtildeath) [reply addons/account/models/account_reconcile_model.py]: Ok I missed that you used `line_residual` in the definition of `total_residual_to_compare` and was left wondering :sweat_smile: 
Changing it now :wink:

**Comment by smetl** [comment]
@grindtildeath Can you change the commit title by
```[FIX] account: Fix matching rules reconciliation when dealing with negative amounts``` plz? Thx :)

**Comment by grindtildeath** [comment]
@smetl Commit amended :)

**Comment by smetl** [comment]
@robodoo r+

**Pull request closed without merging**